plugin/pkg/admission/storage/persistentvolume/label/admission_test.go

```diff
@@ -267,4 +305,54 @@ func TestAdmission(t *testing.T) {
 			t.Errorf("NodeSelectorRequirement %s-in-%v not found in volume NodeAffinity", k, v)
 		}
 	}
+
+	// Labels for GCE Persistent Disks should be applied if there's a GCEPersistentDisk
+	gceConfig := &gce.CloudConfig{
+		ProjectID:          "testprojid",
+		NetworkProjectID:   "testnetprojid",
+		Region:             "testreg",
+		Zone:               "testzone",
+		ManagedZones:       []string{"test"},
+		NetworkName:        "testnet",
+		SubnetworkName:     "testsubnet",
+		SecondaryRangeName: "testsecondrange",
+		NodeTags:           []string{},
+		UseMetadataServer:  false,
+		TokenSource:        &gceMockTokenSource{},
+	}
+	mgce := gcecloud.NewMockGCE(nil)
+	mgce.MockDisks.GetHook = gceMockDisksGetHook
+	mgce.MockBetaRegionDisks.GetHook = gceMockBetaRegionDisksGetHook
+	gceProvider, err := gce.CreateGCECloudWithCloud(gceConfig, mgce)
+	if err != nil {
+		t.Errorf("Error creating the GCE Cloud provider mock: %v", err)
+	}
+	pvHandler.gceCloudProvider = gceProvider
+	err = handler.Admit(admission.NewAttributesRecord(&gcePV, nil, api.Kind("PersistentVolume").WithVersion("version"), gcePV.Namespace, gcePV.Name, api.Resource("persistentvolumes").WithVersion("version"), "", admission.Create, nil))
+
+	if err != nil {
+		t.Errorf("Expected no error when creating gce pv")
+	}
+
+	gcelabels, err := pvHandler.gceCloudProvider.GetLabelsForVolume(context.TODO(), &corev1.PersistentVolume{
+		Spec: corev1.PersistentVolumeSpec{
```

Comment:
Looks like the test added in this PR and verifications below is geared to test `GetLabelsForVolume` in `GCECloud`. So this test should be moved to [gce_disk_test](https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/gce/gce_disks_test.go)?

Tests here should analyze and verify the contents of the `labels` and `NodeAffinity` members of `gcePV` after it has been subjected to `handler.Admit()`

Reply:
The purpose of this pr is to analyze the content of the labels after this function is called https://github.com/fntlnz/kubernetes/blob/6b5f4ebb771e2317e1a62e5bb0594bacf5e71f8d/plugin/pkg/admission/storage/persistentvolume/label/admission.go#L227-L250 - the content only changes when a GCE disk is attached.

Reply:
Clarified a bit more on slack. A test in `TestAdmission` for GCE should verify labels + nodeaffinity on `gcePV`. A test for `findGCEPDLabels` should go under a separate `TestFindGCEPDLabels` function (along with a similar `TestFindEBSLabels` which may come in later).